base/files/file_util_starboard.cc

```diff
@@ -320,7 +320,8 @@ bool CreateDirectoryAndGetError(const FilePath &full_path, File::Error* error) {
 
   // Fast-path: can the full path be resolved from the full path?
   if (DirectoryExists(full_path) ||
-      SbDirectoryCreate(full_path.value().c_str())) {
+      mkdir(full_path.value().c_str(), 0700) == 0 ||
+      SbDirectoryCanOpen(full_path.value().c_str())) {
```

Comment:
If mkdir fails, why would we need to check if we can open the directory?

Reply:
mkdir fails if the directory is already created. In the implementation of SBDirectoryCreate in https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/COBALT:starboard/shared/posix/directory_create.cc;l=73 , we check to see if the directory already exists in the event that it fails.
I could also swap the two conditionals to make it clearer (exists or created), but in the linked implementation, we call SbDirectoryCanOpen at the end to check for race conditions.

Reply:
My point here is that we already check `DirectoryExists` above—does that not make the `SbDirectoryCanOpen` call redundant? (the other uses of SbDirectoryCanOpen all look fine)

Reply:
Oh yeah you're right; fixing that now.